index.js

```diff
@@ -1,125 +1,230 @@
+const GlobSync = glob.GlobSync
+
+const twoStar = '(?:(?!(?:/|^)\\.).)*?' // match "**"
+
+function makeSubpathExpressions (mm) {
+  const set = mm.set
+  const slen = set.length
+  const result = []
+  for (let s = 0; s < slen; s++) {
+    const sn = (s + 1) % slen // next or first exp
+    const exp = set[s]
+    const parts = []
+
+    let endParen = -1
+    for (let e = 0; e < exp.length; e++) {
+      let src
+      let isDynamic = true
+      const subexp = exp[e]
+      if (subexp === GLOBSTAR) {
+        src = twoStar
+      } else if (typeof subexp !== 'string') {
+        src = subexp.source.slice(1, -1)
+      } else {
+        src = regExpEscape(subexp)
+        // if /{a,b}/ is used isDynamic will be true
+        isDynamic = (exp[e] !== set[sn][e])
+      }
+      if (isDynamic) {
+        if (endParen < 0) {
+          src = '(' + src
+        }
+        endParen = e
+      }
+      parts.push(src)
+    }
+    parts[endParen] += ')'
+
+    let extname
+    const last = exp.length - 1
+    if (typeof exp[last] !== 'string' || exp[last] !== set[sn][last]) {
+      // grab extension from filename
+      extname = mm.globParts[s][last].match(/(?:\.[A-Za-z0-9]+)*$/)[0]
+    }
+
+    result.push({
+      regexp: new RegExp('^' + parts.join('/') + '$', 'i'),
+      extname
+    })
   }
-  if (pattern.startsWith('/')) {
-    throw path.buildCodeFrameError(`Glob pattern must be relative, was '${pattern}'`)
+  return result
+}
+
+function regExpEscape (s) {
+  return s.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&')
+}
+
+function generateMembers (gm, cwd) {
+  const expressions = makeSubpathExpressions(gm.minimatch)
+  return gm.found.map(file => {
+    let subpath
+    for (const exp of expressions) {
+      const match = file.match(exp.regexp)
+      if (match) {
+        if (exp.extname) {
+          subpath = match[1].slice(0, -exp.extname.length)
+        } else {
+          subpath = match[1]
+        }
+        break
+      }
+    }
+    return {
+      file,
+      relative: './' + relative(cwd, resolve(cwd, file)),
+      name: memberify(subpath)
+    }
+  })
+}
+
+function memberify (subpath) {
+  const pieces = subpath.split(fileSeparator)
+  const prefixReservedWords = pieces.length === 1
+  const ids = []
+  for (let i = 0; i < pieces.length; i++) {
+    const name = pieces[i]
+    const id = identifierfy(name, {
+      prefixReservedWords,
+      prefixInvalidIdentifiers: i === 0
+    })
+    if (id === null) {
+      return null
+    }
+    ids.push(id)
   }
-  return pattern
+  return ids.join('$')
 }
 
-function rejectImportDefaultSpecifier (path, specifiers) {
-  for (const specifier of specifiers) {
-    if (specifier.type === 'ImportDefaultSpecifier') {
-      throw path.buildCodeFrameError('Cannot import the default member')
+function hasImportDefaultSpecifier (specifiers) {
+  for (const s of specifiers) {
+    if (s.type === 'ImportDefaultSpecifier') {
+      return true
     }
   }
+  return false
```

Comment:
I suppose this could be `return specifiers.some(s => s.type === 'ImportDefaultSpecifier')`.